dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,61 +249,54 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
-    // Retrieve DisplayName and IconUri
-    // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-    // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
-    winrt::com_ptr<IPropertyStore> propertyStore;
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
-    THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
-
     // Retrieve the display name
     std::wstring displayName{};
+    displayName = SetDisplayNameBasedOnProcessName();
 
-    wil::unique_prop_variant propVariantDisplayName;
-    // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
-    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
-    if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
-    {
-        displayName = propVariantDisplayName.pwszVal;
-    }
+    // Retrieve the icon
+    std::wstring iconFilePath{};
 
-    if (displayName.length() == 0)
+    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+
+    if (hWindow)
     {
-        THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, displayName));
+        // Retrieve DisplayName and IconUri
+        // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
+        // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
+        winrt::com_ptr<IPropertyStore> propertyStore;
+        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
+
+        wil::unique_prop_variant propVariantDisplayName;
+        // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
+        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
+        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
+        {
+            displayName = propVariantDisplayName.pwszVal;
+        }
 
-        size_t lastBackslashPosition{ displayName.rfind(L"\\") };
-        THROW_HR_IF(E_UNEXPECTED, lastBackslashPosition == std::wstring::npos);
-        displayName = displayName.substr(lastBackslashPosition + 1); // One after the delimiter
+        wil::unique_prop_variant propVariantIcon;
+        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
+        THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt == VT_EMPTY, "You must specify a valid image filepath before calling Register().");
 
-        displayName.erase(displayName.find_first_of(L".")); // Remove file extension
-    }
+        iconFilePath = propVariantIcon.pwszVal;
```

Comment:
Inconsistency in pattern: You check for this above but not here:
propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr

Reply:
@danielayala94 why just check for VT_EMPTY in filepath we should be doing this instead:

 THROW_HR_IF_MSG(E_UNEXPECTED, propVariantDisplayName.vt != VT_LPWSTR", You must specify a valid image filepath before calling Register().");

Let me know what you think?